src/seedu/addressbook/AddressBook.java

```diff
@@ -383,12 +394,88 @@ private static String executeCommand(String userInputString) {
         case COMMAND_EXIT_WORD:
             executeExitProgramRequest();
+            //Exits program, no return expected.
+        case COMMAND_SORT_WORD:
+        	return executeSortAndListAllPersonsInAddressBook();
+        case COMMAND_EDIT_WORD:
+        	return executeEditPerson(commandArgs);
         default:
             return getMessageForInvalidCommandInput(commandType, getUsageInfoForAllCommands());
         }
     }
 
     /**
+     * Edit a person based on index shown in last list/find.
+     * @param commandArgs
+     * @return
+     */
+    private static String executeEditPerson(String commandArgs) {
+    	String[] editPersonArgs = splitCommandWordAndArgs(commandArgs);
+    	String editPersonIndex = editPersonArgs[0];
+    	String editedPersonDetails = editPersonArgs[1];
+    	
+    	if (!isPersonIndexArgsValid(editPersonIndex)) {
+            return getMessageForInvalidCommandInput(COMMAND_EDIT_WORD, getUsageInfoForEditCommand());
+        }
+    	
+        final int targetVisibleIndex = extractTargetIndexFromPersonIndexArgs(editPersonIndex);
+        
+        if (!isDisplayIndexValidForLastPersonListingView(targetVisibleIndex)) {
+            return MESSAGE_INVALID_PERSON_DISPLAYED_INDEX;
+        }
+        
+        final String[] targetInModel = getPersonByLastVisibleIndex(targetVisibleIndex);
+        
+        return editPersonFromAddressBook(targetInModel, editedPersonDetails);
+	}
+
+	private static final String EDITED_RESULT_DESC = "Successfully Edited";
+
+	/**
+     * Write edited information to selected person
+     * @param targetInModel, editPersonDetails
+     * @return result of edit.
+     */
+	private static String editPersonFromAddressBook(String[] targetInModel, String editedPersonDetails) {
+		Optional<String[]> decodedPerson = decodePersonFromString(editedPersonDetails);
+		
+		// checks if args are valid (decode result will not be present if the person is invalid)
+        if (!decodedPerson.isPresent()) {
+            return getMessageForInvalidCommandInput(COMMAND_EDIT_WORD, getUsageInfoForEditCommand());
+        }
+        
+        deletePersonFromAddressBook(targetInModel);
+        addPersonToAddressBook(decodedPerson.get());
+        
+        return EDITED_RESULT_DESC;
+	}
+
+	/**
+     * Lists the persons in the address book in alphabetical order.
+     * @return text to indicate sorting has completed
+     */
+    private static final String SORTED_RESULT_DESC = "Sorted";
+
+    private static String executeSortAndListAllPersonsInAddressBook() {
+    	ArrayList<String[]> toSort = getAllPersonsInAddressBook();
+    	sort(toSort);
+    	showToUser(toSort);
+    	return SORTED_RESULT_DESC;
+	}
+
+	private static void sort(List<String[]> toSort) {
+		Collections.sort(toSort, new Comparator<String[]>(){
+
+			@Override
+			public int compare(String[] person1, String[] person2) {
+				return person1[0].toLowerCase().compareTo(person2[0].toLowerCase());
+			}
+    		
+    	});
+	}
```

Comment:
Please fix the indentation problem. Other than that, most of your code follows our coding standard.